File: eval/corpus/go/circle.go
package p

import "math"

// Circle is a Shaper implementer: a round shape parameterized by its radius.
type Circle struct {
	radius float64
}

// Area returns the area of the circle, pi*r^2. This satisfies the Shaper interface,
// so a call on a Shaper-typed receiver can dispatch here.
func (c Circle) Area() float64 {
	return math.Pi * c.radius * c.radius
}

// Perimeter returns the circumference of the circle.
func (c Circle) Perimeter() float64 {
	return 2 * math.Pi * c.radius
}

File: eval/corpus/cpp/circle.cpp
#include "shape.h"

// Circle is a concrete Shape implementer: it overrides the abstract area()/name().
class Circle : public Shape {
public:
  explicit Circle(double r) : radius(r) {}
  double area() const override;
  const char *name() const override { return "circle"; }
private:
  double radius;
};

double Circle::area() const {
  return 3.14159 * radius * radius;
}
